packages/sdk-demo/src/context/themeContext/themeContext.tsx

```diff
@@ -0,0 +1,107 @@
+import {
+  useState,
+  createContext,
+  useContext,
+  ReactNode,
+  useEffect,
+} from 'react';
+import { useLocalStorage } from 'react-use';
+
+import { ThemeOptions } from '@mui/material';
+import {
+  materialLight as themeMaterialLight,
+  materialDark as themeMaterialDark,
+  moniteLight as themeMoniteLight,
+  moniteDark as themeMoniteDark,
+} from '@team-monite/sdk-themes';
+
+interface ThemeContextProviderProps {
+  children: ReactNode;
+}
+
+interface ThemeContextValue {
+  themeIndex: 'material' | 'monite';
+  theme: ThemeOptions;
+  colorMode: 'light' | 'dark';
+  setTheme: (theme: 'material' | 'monite') => void;
+  toggleColorMode: () => void;
+}
+
+interface LocalStorageTheme {
+  themeIndex: 'material' | 'monite';
+  colorMode: 'light' | 'dark';
+}
+
+const getTheme = (
+  theme: 'material' | 'monite',
+  colorMode: 'light' | 'dark'
+) => {
+  if (theme === 'material') {
+    return colorMode === 'light' ? themeMaterialLight : themeMaterialDark;
+  }
+
+  return colorMode === 'light' ? themeMoniteLight : themeMoniteDark;
+};
+
+export const ThemeContext = createContext<ThemeContextValue>({
+  themeIndex: 'material',
+  theme: themeMaterialLight,
+  colorMode: 'light',
+  setTheme: () => {},
+  toggleColorMode: () => {},
+});
+
+export const ThemeContextProvider = ({
+  children,
+}: ThemeContextProviderProps) => {
+  const [localStorageTheme, setLocalStorageTheme] =
+    useLocalStorage<LocalStorageTheme>('theme');
```

Comment:
A few notes:
- `useLocalStorage(key, initialData)` has `initialData` argument
- Next.js can't access Local Storage during SSR. Probably, we gonna make server-side requests to retrieve metadata with the theme config from Clerk's private meta data (why not?)

We need to add custom handlers to the provider:

```tsx
<ThemeContextProvider
  onChange={save_To_Local_Storage_Or_To_The_Clerk}
  value={initialTheme_FromLocal_Storage_or_Backend_Request}
/>
```

Reply:
added